roles/confd/tasks/main.yml

```diff
@@ -96,34 +96,46 @@
     - name: Prepare haproxy.tmpl template file (replace "bind" for stats)
       ansible.builtin.lineinfile:
         path: /etc/confd/templates/haproxy.tmpl
-        regexp: "{{ item.regexp }}"
-        line: "{{ item.line }}"
+        regexp: "{{ line_item.regexp }}"
+        line: "{{ line_item.line }}"
         backrefs: true
       loop:
-        - { regexp: '^.*bind.*:{{ haproxy_listen_port.stats }}$', line: '    bind {{ hostvars[inventory_hostname].inventory_hostname }}:{{ haproxy_listen_port.stats }}' }
-        - { regexp: '^.*bind.*:{{ haproxy_listen_port.master }}$', line: '    bind {{ cluster_vip }}:{{ haproxy_listen_port.master }}' }
-        - { regexp: '^.*bind.*:{{ haproxy_listen_port.replicas }}$', line: '    bind {{ cluster_vip }}:{{ haproxy_listen_port.replicas }}' }
-        - { regexp: '^.*bind.*:{{ haproxy_listen_port.replicas_sync }}$', line: '    bind {{ cluster_vip }}:{{ haproxy_listen_port.replicas_sync }}' }
-        - { regexp: '^.*bind.*:{{ haproxy_listen_port.replicas_async }}$', line: '    bind {{ cluster_vip }}:{{ haproxy_listen_port.replicas_async }}' }
+        - regexp: '^.*bind.*:{{ haproxy_listen_port.stats }}$'
+          line: '    bind {{ hostvars[inventory_hostname].inventory_hostname }}:{{ haproxy_listen_port.stats }}'
+        - regexp: '^.*bind.*:{{ haproxy_listen_port.master }}$'
+          line: '    bind {{ cluster_vip }}:{{ haproxy_listen_port.master }}'
+        - regexp: '^.*bind.*:{{ haproxy_listen_port.replicas }}$'
+          line: '    bind {{ cluster_vip }}:{{ haproxy_listen_port.replicas }}'
+        - regexp: '^.*bind.*:{{ haproxy_listen_port.replicas_sync }}$'
+          line: '    bind {{ cluster_vip }}:{{ haproxy_listen_port.replicas_sync }}'
+        - regexp: '^.*bind.*:{{ haproxy_listen_port.replicas_async }}$'
+          line: '    bind {{ cluster_vip }}:{{ haproxy_listen_port.replicas_async }}'
       loop_control:
-        label: "{{ item.line }}"
+        loop_var: line_item
+        label: "{{ line_item.line }}"
       notify: "restart confd"
-      when: cluster_vip is defined and cluster_vip | length > 0
+      when: cluster_vip is defined and cluster_vip | length > 0 hostvars[inventory_hostname].inventory_hostname
```

Comment:
> when: cluster_vip is defined and cluster_vip | length > 0 hostvars[inventory_hostname].inventory_hostname

please tell me why "hostvars[inventory_hostname].inventory_hostname" was added here

Reply:
Apologies for the oversight on my part, it was a copying error which I will rectify

Just a small note of attention, this pull request is still in draft mode as I am currently in the process of refactoring (following a TDD approach)

Nonetheless, thank you for your vigilance and feedback. It is indeed the positive aspect of the review process

Reply:
Yes, I promise to do a careful review when the PR is ready. I was just interested in looking at the progress of the work so that we could discuss a few details at an early stage. Thanks!

Reply:
fixed